src/components/VRating/VRating.ts

```diff
@@ -0,0 +1,181 @@
+import '../../stylus/components/_rating.styl'
+
+import { VNode, CreateElement, VNodeChildrenArrayContents } from 'vue'
+
+import mixins from '../../util/mixins'
+
+// Components
+import VIcon from '../VIcon'
+
+// Mixins
+import Colorable from '../../mixins/colorable'
+import Sizeable from '../../mixins/sizeable'
+import Rippleable from '../../mixins/rippleable'
+
+import { createRange, remapInternalIcon } from '../../util/helpers'
+
+type ItemSlotProps = {
+  index: number
+  value: number
+  isFilled: boolean
+  isHalfFilled: boolean
+  isHovered: boolean
+  isHalfHovered: boolean
+  click: Function
+}
+
+/* @vue/component */
+export default mixins(Colorable, Sizeable, Rippleable).extend({
+  name: 'v-rating',
+
+  props: {
+    backgroundColor: {
+      type: String,
+      default: 'secondary'
+    },
+    color: {
+      type: String,
+      default: 'primary'
+    },
+    emptyIcon: {
+      type: String,
+      default: '$vuetify.icons.ratingEmpty'
+    },
+    fullIcon: {
+      type: String,
+      default: '$vuetify.icons.ratingFull'
+    },
+    halfIcon: {
+      type: String,
+      default: '$vuetify.icons.ratingHalf'
+    },
+    halfIncrements: Boolean,
+    length: {
+      type: Number,
+      default: 5
+    },
+    readonly: Boolean,
+    showHover: Boolean,
+    value: {
+      type: Number,
+      required: true
+    }
+  },
+
+  data: () => ({
+    hoverIndex: -1
+  }),
+
+  computed: {
+    isHovering (): boolean {
+      return this.showHover && this.hoverIndex >= 0
+    }
+  },
+
+  methods: {
+    isHalfEvent (e: MouseEvent): boolean {
+      if (this.halfIncrements) {
+        const rect = e.target && (e.target as HTMLElement).getBoundingClientRect()
+        if (rect && e.offsetX < rect.width / 2) return true
+      }
+
+      return false
+    },
+    createClickFn (i: number): Function {
+      return (e: MouseEvent) => {
+        if (this.readonly) return
+
+        if (this.isHalfEvent(e)) this.$emit('input', i + 0.5)
+        else this.$emit('input', i + 1)
+      }
+    },
+    createProps (i: number): ItemSlotProps {
+      const click = this.createClickFn(i)
+      const isHovered = Math.floor(this.hoverIndex) > i
+      const isHalfHovered = this.halfIncrements && !isHovered && (this.hoverIndex - i) % 1 > 0
+      const isFilled = Math.floor(this.value) > i
+      const isHalfFilled = this.halfIncrements && !isFilled && (this.value - i) % 1 > 0
+
+      return {
+        index: i,
+        value: this.value,
+        isFilled,
+        isHalfFilled,
+        isHovered,
+        isHalfHovered,
+        click
+      }
+    },
+    getIconName (props: ItemSlotProps): string {
+      let iconName
+
+      if (this.isHovering) {
+        iconName = props.isHovered ? this.fullIcon : props.isHalfHovered ? this.halfIcon : this.emptyIcon
+      } else {
+        iconName = props.isFilled ? this.fullIcon : props.isHalfFilled ? this.halfIcon : this.emptyIcon
+      }
+
+      return remapInternalIcon(this, iconName)
```

Comment:
Isn't this still done by v-icon anyway? Why not return the raw value and let that sort itself out?